libsolidity/ast/AST.h

```diff
@@ -1241,6 +1245,12 @@ class EventDefinition: public CallableDeclaration, public StructurallyDocumented
 	{
 	}
 
+	struct CompareBySignature
```

Comment:
I was initially going for a lambda, but then I wanted to use  `eventSignature` inside the comparison function and ended up creating the `struct` here. I agree it is overkill.

> You're using it only once in one function for sorting the container you're iterating over. Why not use ranges::sort() instead, with a lambda function for a comparator?

I had the purpose of using it rather to avoid duplicate signatures than to sort the events. I am assuming that the events in `definedInterfaceEvents()` will be inserted in the resulting `set` prior to those in `usedInterfaceEvents()`. That would guarantee that if there are events with the same signature, only those defined in the contract (or inherited by) are kept in the `set`. Does that make sense? 

https://github.com/ethereum/solidity/blob/e44cdc728b48018aa0d1b415f8a1b3095e51206d/libsolidity/interface/Natspec.cpp#L81-L84

Reply:
I refactored the code to use just lambda functions and encapsulated everything inside a method of `ContractDefinition` class so it can be used for both userdoc and devdoc. 
I think that is better than the previous implementation.

Reply:
It's better now, but see https://github.com/ethereum/solidity/pull/10996#discussion_r1180534169.